lang/py/src/avro/io.py

```diff
@@ -119,11 +120,27 @@ def validate(expected_schema, datum):
       return isinstance(datum, Decimal)
     return isinstance(datum, str)
   elif schema_type == 'int':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
+      if expected_schema.logical_type == 'date':
+        return isinstance(datum, datetime.date)
+      elif expected_schema.logical_type == 'time-millis':
```

Comment:
even in java [https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java#L98](value) corresponding to the constant `time_millis` is `time-millis`